File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int mx=0;
        int ans=0;
        int n=values.size();
        for(int i=0;i<n;i++){
            ans=max(ans,mx+values[i]-i);
            mx=max(mx,values[i]+i);
        }
        return ans;
    }
};

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        //int mxi=0;
        int mx = values[0];
        int n = values.length;
        // for(int i =1;i<n-1;i++){
        //     //System.out.println(i+values[i]);
        //     if(i+values[i]>=mx){
        //         mx = i+values[i];
        //         mxi =i;
        //     }
        // }
        // System.out.println(mxi);
        // int mxj =1;
        // mx = values[1] -1;
        int ans = 0;
        for(int i =1;i<n;i++){
            System.out.println(values[i] - i);
            
            ans = Math.max(ans,mx+values[i] - i);
            mx = Math.max(values[i] + i,mx);
        }
        //System.out.println(mxj);
        return ans;
    }
}